core/arch/arm/kernel/embedded_ts.c

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * Copyright (c) 2020, Arm Limited.
+ */
+#include <crypto/crypto.h>
+#include <initcall.h>
+#include <kernel/embedded_ts.h>
+#include <kernel/ts_store.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <trace.h>
+#include <utee_defines.h>
+#include <util.h>
+#include <zlib.h>
+
+struct ts_store_handle {
+	const struct embedded_ts *ts;
+	size_t offs;
+	z_stream strm;
+};
+
+static void *zalloc(void *opaque __unused, unsigned int items,
+		    unsigned int size)
+{
+	return malloc(items * size);
+}
+
+static void zfree(void *opaque __unused, void *address)
+{
+	free(address);
+}
+
+static bool decompression_init(z_stream *strm,
+			       const struct embedded_ts *ts)
+{
+	int st;
+
+	strm->next_in = ts->ts;
+	strm->avail_in = ts->size;
+	strm->zalloc = zalloc;
+	strm->zfree = zfree;
+	st = inflateInit(strm);
+	if (st != Z_OK) {
+		EMSG("Decompression initialization error (%d)", st);
+		return false;
+	}
+
+	return true;
+}
+
+TEE_Result emb_ts_open(const TEE_UUID *uuid,
+		       struct ts_store_handle **h,
+		       const struct embedded_ts* (*find_ts)
+		       (const TEE_UUID *uuid))
+{
+	struct ts_store_handle *handle;
+	const struct embedded_ts *ts;
+	bool st;
+
+	ts = find_ts(uuid);
+	if (!ts)
+		return TEE_ERROR_ITEM_NOT_FOUND;
+
+	handle = calloc(1, sizeof(*handle));
+	if (!handle)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	if (ts->uncompressed_size) {
+		st = decompression_init(&handle->strm, ts);
+		if (!st) {
```

Comment:
suggesting
```c
		if (!decompression_init(&handle->strm, ts)) {
			free(handle);
			return TEE_ERROR_BAD_FORMAT;
		}
```
Prevents having 2 local vars names `ts` and `st`.

Reply:
done